File: 1688-count-of-matches-in-tournament/1688-count-of-matches-in-tournament.py
class Solution:
    def numberOfMatches(self, n: int) -> int:
        rtn=0
        c=0
        while n!=1:
            
            if (n&1)==0:
                rtn=n//2
                n=rtn
            else:
                rtn=(n-1)//2
                n=rtn +1
            
            c+=rtn
        return c;
            
            
        

File: 1688-count-of-matches-in-tournament/1688-count-of-matches-in-tournament.java
class Solution {
    public int numberOfMatches(int n) {
          
        int c=0,rtn=0;
        while(n !=1 ){
            if((n&1)==0){
                rtn=n/2;
                n=rtn;
            }
            else{
                rtn=(n-1)/2;
                n=rtn+1;
            }
            c+=rtn;
        }
        
        return c;
        
    }
}